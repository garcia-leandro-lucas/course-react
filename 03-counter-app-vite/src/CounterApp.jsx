import {useState} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState(value);

    const handleAdd = (e) => {
       setCounter((count) => count + 1)
    } 

    const handleTakeOut = (e) => {
        setCounter((count) => count - 1)
    } 

    const reset = (e) => {
        setCounter((count) => count = value)
    } 

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={handleTakeOut}>-1</button>
            <button aria-label="btn-reset" onClick={reset}>Reset</button>
        </>
    );

}

CounterApp.propTypes = {
    value: PropTypes.number
};

