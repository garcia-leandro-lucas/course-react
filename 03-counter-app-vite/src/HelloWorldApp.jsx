/* function App() {
    //document.createElement...
    return <h1>Hola Mundo</h1>;
}

export default App; */

/* export function HelloWorldApp() {
    //document.createElement...
    return <h1>Hola Mundo</h1>;
} */


export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  )
}

//La misma funcion de arriba en una sola linea.
/* export const HelloWorldApp = () => <h1>Hello World App</h1>; */
