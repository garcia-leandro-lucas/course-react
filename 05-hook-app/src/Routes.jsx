import { createBrowserRouter, Navigate } from "react-router-dom";
import { AboutPage, HomePage, LoginPage, MainApp, ErrorPage } from "./09-useContext/index";

export const getRoutes = () => createBrowserRouter([
    {
        path: "/",
        element: <MainApp />,
        /* errorElement: <ErrorPage />, */
        children: [
            {
                path: "/",
                element: <HomePage />,
            },
            {
                path: "login",
                element: <LoginPage />,
            },
            {
                path: "about",
                element: <AboutPage />,
            },
            {
                path: "404",
                element: <ErrorPage />,
            },
            {
                path: "*",
                element: <Navigate to="404"/>
            }
        ]
    }
]);