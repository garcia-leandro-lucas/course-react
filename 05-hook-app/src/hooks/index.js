export * from './useFetch';
export * from './useCounter';
export * from './useTodo'