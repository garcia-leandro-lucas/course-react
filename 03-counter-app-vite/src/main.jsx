import React from 'react';
import ReactDom from 'react-dom/client'

/* import {HelloWorldApp} from './HelloWorldApp'; */

import {FirstApp} from './FirstApp'
import { CounterApp } from './CounterApp'

import './index.css';

ReactDom.createRoot(document.getElementById('root')).render(

    <React.StrictMode>
        <CounterApp value={200}/>
        {/* <FirstApp title="Hola soy Vegeta" /> */}
    </React.StrictMode>

);




