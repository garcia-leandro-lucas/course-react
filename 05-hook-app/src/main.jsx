import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { RouterProvider } from 'react-router-dom';

import './index.css'
/* import { CounterApp } from './01-useState/CounterApp' */
/* import { HooksApp } from './HooksApp'; */
/* import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'; */
/* import { SimpleForm } from './02-useEffect/SimpleForm'; */
/* import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'; */
/* import { MultipleCustomHook } from './03-examples/MultipleCustomHook'; */
/* import { FocusScreen } from './04-useRef/FocusScreen'; */
/* import { Layout } from './05-useLayoutEffect/Layout'; */
/* import { Memorize } from './06-memos/Memorize'; */
/* import { MemoHook } from './06-memos/MemoHook'; */
/* import { CallBackHook } from './06-memos/CallBackHook'; */
/* import { Padre } from './07-tarea-memo/Padre'; */

/* import './08-useReducer/intro-reducer'; */
/* import { TodoApp } from './08-useReducer/TodoApp'; */

import { getRoutes } from './Routes';

const router = getRoutes();

createRoot(document.getElementById('root')).render(
  <StrictMode>
    {/*  <HooksApp /> */}
    {/*  <CounterApp /> */}
    {/* <CounterWithCustomHook /> */}
    {/* <SimpleForm /> */}
    {/* <FormWithCustomHook /> */}
    {/* <MultipleCustomHook /> */}
    {/* <FocusScreen /> */}
    {/* <Layout /> */}
    {/* <Memorize /> */}
    {/* <MemoHook /> */}
    {/* <CallBackHook /> */}
    {/* <Padre /> */}
    {/* <TodoApp /> */}
    <RouterProvider router={router}/>
  </StrictMode>
)
