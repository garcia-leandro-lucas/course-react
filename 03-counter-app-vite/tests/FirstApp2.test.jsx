import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en FirstApp.jsx', () => { 

    const title = "Hola, soy un title por fuera del test";
    const subTitle = "Hola, soy un Subtitle por fuera del test";

    test('Debe de hacer match con snapshot', () => { 


        const { container } = render( <FirstApp title={title} /> );
        expect( container ).toMatchSnapshot();

    })

    test('Debe de mostrar el mensaje "Hola Soy Goku', () => { 

        render( <FirstApp title={ title } />);

        // Para negar el toBeTruthy() hacemos un .not
        /* expect( screen.getByText(title) ).not.toBeTruthy(); */

        expect( screen.getByText(title) ).toBeTruthy();

    })


    test('Debe de mostrar el titulo en un h1', () => { 

        render( <FirstApp title={ title } />);
        expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(title);

    });

    test('Debe de mostrar el subtitulo enviado por propds', () => { 

        render( 
        <FirstApp 
            title={ title } 
            subTitle={subTitle}
        />);

        expect( screen.getAllByText(subTitle).length ).toBe(2);

    })



 })