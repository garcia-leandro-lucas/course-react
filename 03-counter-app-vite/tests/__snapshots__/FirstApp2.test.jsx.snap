// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp.jsx Debe de hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy un title por fuera del test
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Leandro
  </p>
</div>
`;
